classes/core/Controllers/Middlewares/Permissions/MatchRoles.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PKP\core\Controllers\Middlewares\Permissions;
+
+use Closure;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Throwable;
+
+class MatchRoles
+{
+    protected function getUserRoles($request): array
+    {
+        if (!$request->user()) {
+            throw new AuthorizationException(
+                'api.403.unauthorized',
+                Response::HTTP_UNAUTHORIZED
+            );
+        }
+
+        $contextId = $request->attributes->get('pkpContext')->getData('id');
+
+        $roles = $request->user()->getRoles($contextId);
+
+        $finalRoles = [];
+        foreach ($roles as $role) {
+            $finalRoles[$role->getId()] = true;
+        }
+
+        return $finalRoles;
+    }
+
+    /**
+     * Verify if current user has the roles
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param string $roles Comma-separated list with roles
+     */
+    public function handle($request, Closure $next, string ...$roles)
+    {
+        try {
+            $userRoles = $this->getUserRoles($request);
+
+            if (array_intersect($userRoles, $roles) === []) {
```

Comment:
At a glance, I'm not sure this is working... `getUserRoles` will return an array of `roleId => true`, and this function will be provided with a string containing a list of numeric constants, if I'm reading it right.

Reply:
Fixed the return of `getUserRoles`. Right now, he's returning an array with the roles Ids, thus enabling us to use `array_intersect` correctly here.